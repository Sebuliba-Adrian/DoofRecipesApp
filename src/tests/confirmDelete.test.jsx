import React from 'react';
import renderer from 'react-test-renderer';
import { mount, configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ConfirmDelete from '../components/confirmDelete';

const props = {
  theId: "deleteCategoryModel1",
  category: {
    id: 1,
    name: "Test Category",
    description: "This is a test category",
    date_created: "Sun, 11 Feb 2018 11:31:50 GMT"
  },
  selectedCategory: {
    id: 1,
    name: "Test Category",
    description: "This is a test category",
    date_created: "Sun, 11 Feb 2018 11:31:50 GMT"
  },
  recipe: {
    id: 1,
    name: 'Test Category',
    description: 'This is a test category',
    date_created: 'Sun, 11 Feb 2018 11:31:50 GMT',
  },
  request: jest.fn(),
  viewRecipes: jest.fn()
};

describe("Confirm delete tests", () => {
  configure({ adapter: new Adapter() });
  const confirmDel = mount(<ConfirmDelete {...props} />);
  const instance = confirmDel.instance();

  it('renders correctly', () => {
    const confirmDelete = renderer
      .create(<ConfirmDelete {...props} />)
      .toJSON();
    expect(confirmDelete).toMatchSnapshot();
  });

  it('Triggers delete', () => {
    const button = confirmDel.find("button").at(2);
    button.simulate("click");
    expect(props.request).toHaveBeenCalled();
    confirmDel.setProps({ theId: "deleteItemModel1" });
    button.simulate("click");
    expect(props.request).toHaveBeenCalled();
  });
  it('Triggers cancel delete', () => {
    const wrapper = shallow(<ConfirmDelete {...props} />);
    wrapper.find("#cancel").simulate("click", { stopPropagation() {} });
  });
});
