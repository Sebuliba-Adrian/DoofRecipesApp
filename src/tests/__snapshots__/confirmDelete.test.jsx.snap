// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm delete tests renders correctly 1`] = `
<div
  className="modal fade"
  id="deleteCategoryModel1"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLabel"
        >
          Delete
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        Are you sure you want to delete?
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          id="cancel"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-primary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
